servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/StreamObserverTest.java

```diff
@@ -122,15 +123,17 @@ protected void channelRead0(final ChannelHandlerContext ctx, final Object msg) {
     }
 
     @AfterEach
-    void teardown() {
-        safeSync(() -> serverAcceptorChannel.close().syncUninterruptibly());
-        safeSync(() -> serverEventLoopGroup.shutdownGracefully(0, 0, MILLISECONDS).syncUninterruptibly());
+    void teardown() throws Exception {
+        safeSync(() -> serverAcceptorChannel.close().sync());
+        safeSync(() -> serverEventLoopGroup.shutdownGracefully(0, 0, MILLISECONDS).sync());
         safeClose(client);
     }
 
-    static void safeSync(Runnable runnable) {
+    static void safeSync(Callable<Object> callable) throws Exception {
         try {
-            runnable.run();
+            callable.call();
+        } catch (InterruptedException e) {
+            throw e;
```

Comment:
> 2. Doesn’t safeClose() in HttpsProxyTest also not Propagate the InterruptedException, which might result in timeouts as well?

Good catch! You are right. Also, before re-throwing the `InterruptedException`, we should reset `interrupted` flag:
```java
Thread.currentThread().interrupt();
throw e;
```

Reply:
Good catch, forgot that it resets the value :)